spec/std/hash_spec.cr

```diff
@@ -257,6 +257,25 @@ describe "Hash" do
       a = {1 => 2}
       a.delete(2).should be_nil
     end
+
+    describe "with block" do
+      it "returns the value if a key is found" do
+        a = {1 => 2}
+        a.delete(1).should eq(2)
+      end
+
+      it "executes block if key is not found" do
```

Comment:
I'd test here case when found key has a `nil` value, like in `{ 3 => nil }.delete(3) { 7 }.should be_nil`